ceremonies/src/benchmarking.rs

```diff
@@ -0,0 +1,386 @@
+{
+	while frame_system::Pallet::<T>::block_number() < n {
+		if frame_system::Pallet::<T>::block_number() > 1.into() {
+			frame_system::Pallet::<T>::on_finalize(frame_system::Pallet::<T>::block_number());
+		}
+		let new_timestamp: u64 = (T::BlockNumber::from(GENESIS_TIME) +
+			T::BlockNumber::from(BLOCKTIME) * n)
+			.unique_saturated_into();
+
+		set_timestamp::<T>(T::Moment::from(new_timestamp));
+		pallet_timestamp::Pallet::<T>::on_finalize(frame_system::Pallet::<T>::block_number());
+		frame_system::Pallet::<T>::set_block_number(
+			frame_system::Pallet::<T>::block_number() + 1.into(),
+		);
+		frame_system::Pallet::<T>::on_initialize(frame_system::Pallet::<T>::block_number());
+	}
+}
+
+/// Progress blocks until the phase changes
+fn run_to_next_phase<T: Config>()
+where
+	<T as frame_system::Config>::BlockNumber: From<u64>,
+	<T as pallet_timestamp::Config>::Moment: From<u64>,
+{
+	let phase = encointer_scheduler::Pallet::<T>::current_phase();
+	let mut blocknr = frame_system::Pallet::<T>::block_number();
+	while phase == encointer_scheduler::Pallet::<T>::current_phase() {
+		blocknr += 1.into();
+		run_to_block::<T>(blocknr);
+	}
+}
+
+pub fn set_timestamp<T: Config>(t: T::Moment) {
+	let _ = pallet_timestamp::Pallet::<T>::set(RawOrigin::None.into(), t);
+}
+
+fn get_proof_of_attendance<T: Config>(
+	prover: &sr25519::Pair,
+	cid: CommunityIdentifier,
+) -> ProofOfAttendance<T::Signature, T::AccountId>
+where
+	<T as frame_system::Config>::AccountId: From<sp_core::sr25519::Public>,
+	<T as Config>::Signature: From<sp_core::sr25519::Signature>,
+{
+	let cindex = encointer_scheduler::Pallet::<T>::current_ceremony_index();
+	let prover_account_id = account_id::<T>(&prover);
+	assert_ok!(encointer_balances::Pallet::<T>::issue(
+		cid,
+		&prover_account_id,
+		NominalIncome::from_num(1)
+	));
+	Pallet::<T>::fake_reputation(
+		(cid, cindex - 1),
+		&account_id::<T>(&prover),
+		Reputation::VerifiedUnlinked,
+	);
+	assert_eq!(
+		Pallet::<T>::participant_reputation((cid, cindex - 1), &prover_account_id),
+		Reputation::VerifiedUnlinked
+	);
+	let cindex = encointer_scheduler::Pallet::<T>::current_ceremony_index();
+	IssuedRewards::<T>::insert((cid, cindex - 1), 0, ());
+	let proof = prove_attendance::<T>(prover_account_id, cid, cindex - 1, &prover);
+	proof
+}
+
+pub fn last_event<T: Config>() -> Option<<T as frame_system::Config>::Event> {
+	let events = frame_system::Pallet::<T>::events();
+	if events.len() < 1 {
+		return None
+	}
+	let frame_system::EventRecord { event, .. } = &events[events.len() - 1];
+	Some(event.clone())
+}
+
+benchmarks! {
+	where_clause {
+		where
+		<T as frame_system::Config>::AccountId: From<sp_core::sr25519::Public>,
+		<T as Config>::Signature: From<sp_core::sr25519::Signature>,
+		<T as pallet_timestamp::Config>::Moment: From<u64>,
+		<T as frame_system::Config>::BlockNumber: From<u64>,
+		<T as frame_system::Config>::Event: From<pallet::Event<T>>
+	}
+
+	// register reputable
+	register_participant {
```

Comment:
Did you decide to cover multiple cases here because you don't know which one is the longest path?

Reply:
I am quite confident that all the paths are of similar complexity. We should only go for the worst-case scenario instead of introducing more complexity here.